azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/diagnostic_logger.py

```diff
@@ -0,0 +1,87 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import logging
+from os import makedirs
+from os.path import exists, join
+
+from azure.monitor.opentelemetry.distro.constants import (
+    _CUSTOMER_IKEY,
+    _DIAGNOSTIC_LOG_PATH,
+    _EXTENSION_VERSION,
+    _IS_ON_APP_SERVICE,
+    _env_var_or_default,
+)
+from azure.monitor.opentelemetry.distro.version import VERSION
+
+_OPENTELEMETRY_DIAGNOSTIC_LOGGER_NAME = "opentelemetry"
+_DISTRO_DIAGNOSTIC_LOGGER_NAME = "azure.monitor.opentelemetry.distro"
+_DIAGNOSTIC_LOGGER_FILE_NAME = "applicationinsights-extension.log"
+_OPERATION = "Startup"
+_SITE_NAME = _env_var_or_default("WEBSITE_SITE_NAME")
+_SUBSCRIPTION_ID = _env_var_or_default("WEBSITE_OWNER_NAME")
+_opentelemetry_logger = logging.getLogger(
+    _OPENTELEMETRY_DIAGNOSTIC_LOGGER_NAME
+)
+_distro_logger = logging.getLogger(_DISTRO_DIAGNOSTIC_LOGGER_NAME)
+
+
+class AzureDiagnosticLogger:
+    _initialized = False
+
+    def initialize():
+        if not AzureDiagnosticLogger._initialized:
+            if _IS_ON_APP_SERVICE:
+                format = (
+                    "{"
+                    + '"time":"%(asctime)s.%(msecs)03d", '
+                    + '"level":"%(levelname)s", '
+                    + '"logger":"%(name)s", '
+                    + '"message":"%(message)s", '
+                    + '"properties":{'
+                    + f'"operation":"{_OPERATION}", '
+                    + f'"sitename":"{_SITE_NAME}", '
+                    + f'"ikey":"{_CUSTOMER_IKEY}", '
+                    + f'"extensionVersion":"{_EXTENSION_VERSION}", '
+                    + f'"sdkVersion":"{VERSION}", '
+                    + f'"subscriptionId":"{_SUBSCRIPTION_ID}", '
+                    + '"language":"python"'
+                    + "}"
+                    + "}"
+                )
+                if not exists(_DIAGNOSTIC_LOG_PATH):
```

Comment:
This logic should be handled in the `get`

Reply:
Not sure I understand what you mean. This is just a check to make sure the path exists in the filesystem. Are you saying this logic should be on the constants side or that _DIAGNOSTIC_LOG_PATH should have a getter method unlike the other constants?